docs/int_header/INT_header.md

```diff
@@ -9,83 +9,61 @@ These definitions contain modifications and required fields for Transparent Secu
-
-## IP Header update
-
-### Encapsulation
-
-We want to preserve the routing and other information on the packet, so we will modify the protocol (IPv4) / next header (IPv6), length and IPv4 the header checksum.
-
-Define a new L3 protocol.  The protocol for the INT header will be 63 "Any local network" since the INT header should never be exposed outside of the service providers network.  There is no specification for protocol 63.
-
-IPv4:
-
-* Update the total length to match the new size
-* Set the protocol to the INT protocol number
-* Update the header checksum
-
-IPv6:
+## INT Header
 
-* Set the Next Header to be the identifier for an IPv6 extension header to the INT protocol number.  If existing extension headers are in place, INT will prepend to the first extension header
-* The INT header next header field will be the existing IPv6 next header value
-* The Hdr Ext Len will be updated appropriately
+### INT metadata header (12 bytes)
 
-### IP header shim (4 bytes)
+The hop-by-hop INT header will follow the header as described in section 4.7. INT Hop-by-Hop Metadata Header Format in the current INT specification.
 
-This shim shall be 4 bytes and include the original protocol (IPv4) or next protocol (IPv6) and the length of the INT metadata.
+* INT instructions are encoded as a bitmap in the 16 bit INT Instruction field.
 
-Type: (1 octet) This field indicates the type of INT Header following the shim header. Two Type values are used: one for the hop-by-hop header type and the other for the destination header type.
+  * bit0: 4 octet Switch ID which is unique across the network
 
-Reserved: (1 octet)
+### Per-Hop INT Metadata record (12 bytes)
 
-Length: (1 octet) This is the total length of INT metadata header, INT stack and the shim header in 4-byte words. A non-INT device may read this field and skip over INT headers.
+This metadata will only contain one record and will not be updated on subsequent hops.
 
-Protocol: (1 octet) If IP protocol / next header is used to indicate INT, this field optionally stores the original protocol / next header value. Otherwise, this field is reserved.
+Each metadata record corresponds to a bit filed in the instruction set and is 4 octets long.  Only bit0 is required to be set for transparent secuirty, but other bits can be set as indicatied by the INT specifiction.
 
-### Decapsulation
+* Bit0: Switch ID: Unique identifier for the switch (4 octets)
 
-The original protocol / next header will be restored and the size and checksum will be recalculated as appropriate.  The IP shim header and INT header are also removed.
+On the customer's gateway, the gateway enters its ID as the switch ID and it inserts the originating devices MAC address.
 
-## INT Header
+If the INT header is created on a switch inside the head end.  This occurs when a header is not added at the customer premises.  Two entries are added, one for the gateway device.  In a DOCSIS network, this is the cable modem.
```

Comment:
It might be good to show an example for this case?  The switch has enough knowledge to add the hop information for the gateway?  This seems different enough to specify a bit more.  Is this a special case of INT source  functionality where it adds hop metadata from previous hops?

Reply:
I've updated this in the DS Metadata section.  Thanks for your feedback.